models/access.go

```diff
@@ -163,17 +171,23 @@ func maxAccessMode(modes ...AccessMode) AccessMode {
 }
 
 // FIXME: do cross-comparison so reduce deletions and additions to the minimum?
-func (repo *Repository) refreshAccesses(e Engine, accessMap map[int64]AccessMode) (err error) {
+func (repo *Repository) refreshAccesses(e Engine, accessMap map[*User]AccessMode) (err error) {
```

Comment:
Why is pointer (`*User`) is needed as map key?

Reply:
Because we need quick access to `User.IsRestricted` several lines later (line 183).

Using `*User` as a map key seemed like the least intrusive change.

Reply:
Having a whole struct as map key sucks pretty much

Reply:
It would be better to create type ex:
```go
type userAccessMode {
  User *User
  AccessMode AccessMode
}
```
And use `map[int64]userAccessMode`

Reply:
If I understand correctly, a pointer to a struct as map key means it's memory address is used as key, which is more or less the same as `int64` performance-wise. 

So, aside from the theoretical duplicates one might end up with when building such a map from `User` structs from different sources, it feels .. "slightly ugly, but acceptable" =)


One could introduce:
```
type UserAccess struct {
    User *User
    Mode AccessMode
}
```

and make `accessMap` a `map[int64]UserAccess`. Would that be more acceptable?

Reply:
yes, this your proposed struct would also be good, just don't need to export it as it would be used only internally -> `type userAccess struct`

Reply:
Something like this maybe?